File: 이코테 복습(python, c++, java)/그리디/c++/모험가 길드.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int n;
vector<int>arr;

int main(void) {
	cin >> n;
	for (int i = 0; i < n; i++) {
		int x;
		cin >> x;
		arr.push_back(x);
	}
	sort(arr.begin(), arr.end());

	int group = 0;
	int total = 0;

	for (int i = 0; i < arr.size(); i++) {
		total += 1;
		if (total >= arr[i]) {
			group++;
			total = 0;
		}
	}
	cout << group << endl;

}

File: 이코테 복습(python, c++, java)/그리디/python/모험가 길드.py
n = int(input())
group = list(map(int, input().split()))
group.sort()
group_cnt = 0
total = 0
for i in range(len(group)):
    total += 1
    if total >= group[i]:
        total = 0
        group_cnt += 1

print(group_cnt)
